boilerplate/openshift/golang-osd-operator/csv-generate/csv-generate.sh

```diff
@@ -0,0 +1,118 @@
+    usage
+    exit 1
+fi
+
+# If no override, using the gitlab repo
+if [ -z "$GIT_PATH" ] ; then 
+    GIT_PATH="https://app:@gitlab.cee.redhat.com/service/saas-${operator_name}-bundle.git"
+fi
+
+# Calculate previous version
+SAAS_OPERATOR_DIR="saas-${operator_name}-bundle"
+BUNDLE_DIR="$SAAS_OPERATOR_DIR/${operator_name}/"
+
+if [ "$test_boilerplate_generation" = true ] ; then
+    OPERATOR_NEW_VERSION=$(ls "$BUNDLE_DIR" | sort -t . -k 3 -g | tail -n 1)
+    OPERATOR_PREV_VERSION=$(ls "${BUNDLE_DIR}" | sort -t . -k 3 -g | tail -n 2 | head -n 1)
+    # Generate with boilerplate generator
+    ./boilerplate/openshift/golang-osd-operator/csv-generate/common-generate-operator-bundle.py -o ${operator_name} -d output-comparison -p ${OPERATOR_PREV_VERSION} -n ${operator_commit_number} -c ${operator_commit_hash} -i ${operator_image}
+    echo "CSV generated with boilerplate generate-bundle script"
+    # Preparing yamls for the diff by removing the creation timestamp
+    yq d ${BUNDLE_DIR}/${OPERATOR_NEW_VERSION}/*.clusterserviceversion.yaml 'metadata.annotations.createdAt' > output-comparison/hack_generate.yaml
+    yq d output-comparison/${OPERATOR_NEW_VERSION}/*.clusterserviceversion.yaml 'metadata.annotations.createdAt' > output-comparison/common_generate.yaml
+    # Diff on the filtered files
+    diff output-comparison/hack_generate.yaml output-comparison/common_generate.yaml
+else
+    rm -rf "$SAAS_OPERATOR_DIR"
+    git clone --branch "$operator_channel" ${GIT_PATH} "$SAAS_OPERATOR_DIR"
+    
+    # remove any versions more recent than deployed hash
+    if [[ "$operator_channel" == "production" ]]; then
+        if [ -z "$DEPLOYED_HASH" ] ; then
+            DEPLOYED_HASH=$(
+                curl -s "https://gitlab.cee.redhat.com/service/app-interface/raw/master/data/services/osd-operators/cicd/saas/saas-${_OPERATOR_NAME}.yaml" | \
+                    docker run --rm -i quay.io/app-sre/yq yq r - "resourceTemplates[*].targets(namespace.\$ref==/services/osd-operators/namespaces/hivep01ue1/${_OPERATOR_NAME}.yml).ref"
+            )
+        fi
+    
+        delete=false
+        # Sort based on commit number
+        for version in $(ls $BUNDLE_DIR | sort -t . -k 3 -g); do
+            # skip if not directory
+            [ -d "$BUNDLE_DIR/$version" ] || continue
+    
+            if [[ "$delete" == false ]]; then
+                short_hash=$(echo "$version" | cut -d- -f2)
+    
+                if [[ "$DEPLOYED_HASH" == "${short_hash}"* ]]; then
+                    delete=true
+                fi
+            else
+                rm -rf "${BUNDLE_DIR:?BUNDLE_DIR var not set}/$version"
+            fi
+        done
+    fi
+    
+    OPERATOR_PREV_VERSION=$(ls "$BUNDLE_DIR" | sort -t . -k 3 -g | tail -n 1)
+    # CSV actual generate
+    ./hack/generate-operator-bundle.py -o ${operator_name} -d ${BUNDLE_DIR} -p ${OPERATOR_PREV_VERSION} -n ${operator_commit_number} -c ${operator_commit_hash} -i ${operator_image}
```

Comment:
I think the script is not taking options but ordered parameters in other operators - to be checked and fixed

Reply:
Fixed in latest commit